### PR TITLE
[AIRFLOW-5000] Remove duplicate end_date and reorder template

### PR DESCRIPTION
**Pull request by zhongjiajie**
* Remove the duplicate end_date in template context
* Reorder template context key to make ds together

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5000

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

* Remove the duplicate end_date in template context
* Reorder template context key to make ds together

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by zhongjiajie** [comment]
I'm not sure we need `END_DATE` in template context or not, but I can not find it in our code base and we already have `end_date`(**lowercase**) with the same values

**Comment by BasPH** [comment]
I'm all in for this change, but FYI I suggested to do the same + more variables a while ago which led to a discussion going around in circles: https://github.com/apache/airflow/pull/5010 (mailing list thread: https://lists.apache.org/thread.html/a81b32efc95db41a5bf847ed03b64ae4d3e9c8a2cbeda1bdcf0eb2f5@%3Cdev.airflow.apache.org%3E).

I suggest to merge this PR and tackle the other variables one-by-one instead of one single PR.

**Review by BasPH — APPROVED** [review]

**Comment by zhongjiajie** [comment]
@BasPH Sorry, I don't read all dev mail list and don't know you submit related PR.

But I find out we still have strong opinions on which variable should be remove. But think it safe to remove the duplicate key `end_date` and `END_DATE`

I find out that some summary in mail list
> We start by putting deprecation warnings on tables, latest_date, end_date and END_DATE and remove them in Airflow 2.0.

I think we still have `end_date` in template, so `@deprecation` is not necessary for this PR

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -1199,23 +1199,22 @@ def __repr__(self):
 
         return {
             'dag': task.dag,
+            'dag_run': dag_run,
+            'run_id': run_id,
```
Comment (zhongjiajie) [inline airflow/models/taskinstance.py]: And what do you think this change. At first I think template messy and all date variable should keep together, but I find out your PR like to keep them in alphabetical
Should I change them in alphabetical or keep them? @BasPH

Comment (BasPH) [reply airflow/models/taskinstance.py]: Let's reorder alphabetically and then merge 👍

**Comment by zhongjiajie** [comment]
Reorder template context alphabetically

**Review by BasPH — APPROVED** [review]
Nice work!

**Comment by zhongjiajie** [comment]
Thanks

**Pull request merged**